spring-boot-modules/spring-boot-jackson/src/main/java/com/baeldung/boot/jackson/config/CoffeeConfiguration.java

```diff
@@ -0,0 +1,60 @@
+package com.baeldung.boot.jackson.config;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
+import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+
+import java.time.format.DateTimeFormatter;
+
+@Configuration
+public class CoffeeConfiguration {
+    public static final String dateTimeFormat = "dd-MM-yyyy HH:mm";
+    private LocalDateTimeSerializer localDateTimeSerializer = new LocalDateTimeSerializer(DateTimeFormatter.ofPattern(dateTimeFormat));
+
+    @Bean
+    public MappingJackson2HttpMessageConverter mappingJackson2HttpMessageConverter() {
+        Jackson2ObjectMapperBuilder builder = new Jackson2ObjectMapperBuilder()
+                .serializers(localDateTimeSerializer)
```

Comment:
The indentation here should be 4 spaces not 8. This applies throughout the whole PR.

Reply:
I am not sure what do you want to say, I am using IntelIJ with Eclipse keymap. 
Before public method we have 4 spaces.

Reply:
It is the second line of the indentation which is wrong. In this case it should be:

```
@Configuration
public class CoffeeHttpConverterConfiguration {

    @Bean
    public MappingJackson2HttpMessageConverter mappingJackson2HttpMessageConverter() {
        Jackson2ObjectMapperBuilder builder = new Jackson2ObjectMapperBuilder()
            .serializers(localDateTimeSerializer)
            .serializationInclusion(JsonInclude.Include.NON_NULL);
        return new MappingJackson2HttpMessageConverter(builder.build());
    }
}
```

Likewise for your controller and I think most of the classes:
```
@RestController
public class CoffeeController {

    @GetMapping("/coffee")
    public Coffee getCoffee(@RequestParam(required = false) String brand, @RequestParam(required = false) String name) {
        return new Coffee().setBrand(brand)
            .setDate(LocalDateTime.now())
            .setName(name);
    }
}
```
If you want to break the method signature across two lines, it is OK.

I don't understand what you mean by " I am using IntelIJ with Eclipse keymap."

You can use our eclipse of IntelliJ formatter directly I think:

- https://github.com/eugenp/tutorials/blob/master/eclipse/formatter.xml
- https://github.com/eugenp/tutorials/blob/master/intelliJ/intelliJ-formatter.xml